From 898a7d3fe9c3a7410d4a7d84394c107ded8c9a50 Mon Sep 17 00:00:00 2001
From: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
Date: Fri, 6 Jun 2025 10:40:31 +0800
Subject: [PATCH 008/119] mtk: mt76: mt7996: fix rx chainmask

Since wiphy->available_antennas_tx now accumulates the chainmask of all the
radios under this wiphy (due to single wiphy), use phy->orig_antenna_mask
to check whether the antenna_mask has been changed (by the user).

Signed-off-by: StanleyYP Wang <StanleyYP.Wang@mediatek.com>
---
 mt7996/mt7996.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mt7996/mt7996.h b/mt7996/mt7996.h
index 97b84710..6723df6d 100644
--- a/mt7996/mt7996.h
+++ b/mt7996/mt7996.h
@@ -714,7 +714,7 @@ void mt7996_memcpy_fromio(struct mt7996_dev *dev, void *buf, u32 offset,
 
 static inline u16 mt7996_rx_chainmask(struct mt7996_phy *phy)
 {
-	int max_nss = hweight8(phy->mt76->hw->wiphy->available_antennas_tx);
+	int max_nss = hweight16(phy->orig_antenna_mask);
 	int cur_nss = hweight8(phy->mt76->antenna_mask);
 	u16 tx_chainmask = phy->mt76->chainmask;
 
-- 
2.45.2

